addon/components/menu.js

```diff
@@ -33,6 +34,11 @@ export default class Menu extends Component {
   @action
   close() {
     this.isOpen = false;
+    next(() => {
+      if (this.buttonElement !== null) {
+        this.buttonElement.focus();
```

Comment:
Hey @far-fetched, thanks for the speedy reply. 

The headlessui tests call `assertActiveElement()` to check whether or not an element is active / focussed. I've added it to the `Menu` tests (for consistency with `Listbox` and `Dialogue`)  and this is where the failure occurred. 

`assertActiveElement` simply waits until `document.element` is equal to the element we pass to it. 
``` 
return Qunit.assert.waitFor(() => {
   Qunit.assert.equal(document.activeElement, element, comment);  
});
```
Unfortunately, `waitFor` times out because `document.activeElement` is always `<body></body>`.

Reply:
Ace work, sorry I missed the other PR, your scenario makes perfect sense.

I'm very happy to refactor and handle the re-focus in the way you describe. I guess it's a question for the maintainers, in terms of how far we want to deviate from the React / Vue implementations? 🤷🏻‍♂️

Reply:
I think that we should be exactly the same in terms of test coverage. Tests which I think are related to return focus for menu already exist in [react](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/menu/menu.test.tsx#L2808-L2920) package.
We have [them](https://github.com/GavinJoyce/ember-headlessui/blob/master/tests/integration/components/menu-test.js#L1336-L1337), but commented out.